### PR TITLE
test: thread : test run k_thread_resume on unsuspend thread

### PR DESCRIPTION
**Pull request by yingmingx**
If calling function k_thread_resume() when the thread is not suspend,
it takes no effect. This change improve coverage of function
k_thread_resume() in sched.c

Signed-off-by: Ying ming <mingx.ying@intel.com>

### CODE REVIEW

**Review by ceolin — APPROVED** [review]

**Review by enjiamai — APPROVED** [review]

File: tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c [hunk 1]
```diff
@@ -27,6 +27,9 @@ static void threads_suspend_resume(int prio)
 	k_tid_t tid = k_thread_create(&tdata, tstack, STACK_SIZE,
 				      thread_entry, NULL, NULL, NULL,
 				      create_prio, K_USER, K_NO_WAIT);
+	/* checkpoint: resume an unsuspended thread has no effect */
+	k_thread_resume(tid);
+	zassert_false(last_prio == create_prio, NULL);
```
Comment (nashif) [inline tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c:32]: how is this check related to k_thread_resume()?

Comment (yingmingx) [reply tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c:32]: Thank you for you comment, you're right. This check has no effect here. I'll change my code.

**Review by andyross — APPROVED** [review]

File: tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c [hunk 2]
```diff
@@ -27,6 +27,18 @@ static void threads_suspend_resume(int prio)
 	k_tid_t tid = k_thread_create(&tdata, tstack, STACK_SIZE,
 				      thread_entry, NULL, NULL, NULL,
 				      create_prio, K_USER, K_NO_WAIT);
+	/* checkpoint: resume an unsuspended thread has no effect */
+	k_thread_resume(tid);
+	k_msleep(100);
+	zassert_true(last_prio == create_prio, NULL);
```
Comment (nashif) [inline tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c:33]: still not sure how this assert on priorities is testing that resume has no effect on unsuspended threads.

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c (hunk 2 above)
Comment (yingmingx) [reply tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c:33]: I added the line 30-34 in this existing function.  The main purpose of these code is to make a comparison between line 42-46. The line 36-37 is just to restore the enviroment.
If I changed the line 31 k_thread_resume to fucntion k_thread_suspend. Then the assert on line 33 would fail.
The new added code can also reflect k_thread_resume() and k_thread_suspend() are totally different function. For example, they are not like mouse buttons, but like light switches.

**Review by LeiW000 — APPROVED** [review]

File: tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c (hunk 2 above)
Comment (yingmingx) [reply tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c:33]: @nashif  Could you take another look at it, thanks!

Comment (nashif) [reply tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c:33]: I am still not convinced you are testing what you claim. The priority of the thread does not indicate anything about what happens when you resume a running thread.

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c (hunk 2 above)
Comment (yingmingx) [reply tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c]: I got you point.  It have changed the code use kernel API to get thread state to judge zassert. 
I've created a new testcase. The new testcase must test in kernel mode because get thread state is not allowed in user mode.

**Review by dcpleung — APPROVED** [review]

**Review by enjiamai — APPROVED** [review]
The newly adding test case is in order to verify k_thread_resume() only do things when the previous state of the thread was suspended. Looks fine for me.

**Comment by yingmingx** [comment]
@nashif  The previous judging condition use thread priority is not proper.
I used k_thread_state_str() to assess the thread status before and after runing k_thread_resume().

**Review by nashif — APPROVED** [review]
